src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceContext.java

```diff
@@ -147,7 +158,17 @@ public final boolean wantForkPRs() {
     public final Set<ChangeRequestCheckoutStrategy> forkPRStrategies() {
         return forkPRStrategies;
     }
-
+    /**
+     * Returns the strategies used to notify Github of build status.
+     * @return the strategies used to notify Github of build status.
+     * @since TODO
+     */
+    public final List<AbstractGitHubNotificationStrategy> notificationStrategies() {
+        if (notificationStrategies.isEmpty()) {
+            return Collections.<AbstractGitHubNotificationStrategy>singletonList(new DefaultGitHubNotificationStrategy());
+        }
+        return notificationStrategies;
```

Comment:
Be consistent in returning an immutable list here. IOW `return Collections.unmodifiableList(notificationStrategies);`